src/Arcus.EventGrid.Tests.Integration/Publishing/CloudEventPublishing.cs

```diff
@@ -0,0 +1,196 @@
+            const string licensePlate = "1-TOM-337";
+            const string expectedSubject = "/";
+            var eventId = Guid.NewGuid().ToString();
+            var data = new CarEventData(licensePlate);
+            var rawEventBody = JsonConvert.SerializeObject(data);
+            var cloudEvent = new CloudEvent(CloudEventsSpecVersion.V1_0, "NewCarRegistered", new Uri("http://test-host"), subject: expectedSubject, id: eventId)
+            {
+                Data = data,
+                DataContentType = new ContentType("application/json")
+            };
+
+            IEventGridPublisher publisher = _endpoint.BuildPublisher();
+
+            // Act
+            await publisher.PublishRawCloudEventAsync(cloudEvent.SpecVersion, cloudEvent.Id, cloudEvent.Type, cloudEvent.Source, rawEventBody);
+            TracePublishedEvent(eventId, cloudEvent);
+
+            // Assert
+            var receivedEvent = _endpoint.ServiceBusEventConsumerHost.GetReceivedEvent(eventId);
+            AssertReceivedNewCarRegisteredEvent(eventId, cloudEvent.Type, expectedSubject, licensePlate, receivedEvent);
+        }
+
+        [Fact]
+        public async Task PublishSingleRawEventWithDetailedInfo_WithBuilder_ValidParameters_Succeeds()
+        {
+            // Arrange
+            const string licensePlate = "1-TOM-337";
+            const string expectedSubject = "/";
+            var eventId = Guid.NewGuid().ToString();
+            var data = new CarEventData(licensePlate);
+            var rawEventBody = JsonConvert.SerializeObject(data);
+            var cloudEvent = new CloudEvent(CloudEventsSpecVersion.V1_0, "NewCarRegistered", new Uri("http://test-host"), subject: expectedSubject, id: eventId, DateTime.UtcNow)
+            {
+                Data = data,
+                DataContentType = new ContentType("application/json")
+            };
+
+            IEventGridPublisher publisher = _endpoint.BuildPublisher();
+
+            // Act
+            await publisher.PublishRawCloudEventAsync(
+                cloudEvent.SpecVersion,
+                cloudEvent.Id,
+                cloudEvent.Type,
+                cloudEvent.Source,
+                cloudEvent.Subject,
+                rawEventBody,
+                cloudEvent.Time ?? default(DateTimeOffset));
+            TracePublishedEvent(eventId, cloudEvent);
+
+            // Assert
+            var receivedEvent = _endpoint.ServiceBusEventConsumerHost.GetReceivedEvent(eventId);
+            AssertReceivedNewCarRegisteredEvent(eventId, cloudEvent.Type, cloudEvent.Subject, licensePlate, receivedEvent);
+        }
+
+        private void TracePublishedEvent(string eventId, object events)
+        {
+            _testOutput.WriteLine($"Event '{eventId}' published - {JsonConvert.SerializeObject(events)}");
+        }
+
+        private static void AssertReceivedNewCarRegisteredEvent(string eventId, string eventType, string eventSubject, string licensePlate, string receivedEvent)
```

Comment:
Shouldn't we share this with other test? Maybe add a custom `Assert.ReceivedNewCarRegisteredEvent`

Reply:
Yes, was also thinking that. But I'm also not sure how the testing infrastructure will eventually look like. Like input/output cloudevent/evengridevent schema. So... not sure if we can already fully make this allign.
But I can move it for now.

Reply:
What schema we use to wrap our payload in will not impact how we assert our payload itself, no?

Reply:
No, I changed that too. It uses the general `Event` to assert.

Reply:
LGTM